config/v1/types_oauth.go

```diff
@@ -40,8 +40,31 @@ type OAuthSpec struct {
 	// templates allow you to customize pages like the login page.
 	// +optional
 	Templates OAuthTemplates `json:"templates"`
+	// audit specifies what should be audited in the context of OAuthServer.
+	// +optional
+	Audit OAuthAudit `json:"audit"`
+}
+
+// OAuthAudit specifies the Audit profile in use.
+type OAuthAudit struct {
+	// profile is a simple drop in profile type.
+	// +kubebuilder:default:="WriteLoginEvents"
```

Comment:
I wonder how this works if the profile is unspecified. What would happen if we add the `+optional` tag here?

Reply:
Ok, so should we maybe delete the optional flag from [Audit](https://github.com/openshift/api/pull/1042/files#diff-17be816abc6f9bcc605991fa9b3aebe956afbb04c48c0ff957216993f3728438R45)? If it is optional, the probability that the default value will be autogenerated is minimal.

Reply:
@sttts, could you verify that the thought process of removing `+optional` is right and tell @stlaz and me what would happen on an `+optional`?